conductor-client/src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -143,60 +142,67 @@ class ExternalDatabaseManagementService(
-        )
-
-        val accessCheckCompletion = entityTypeId.thenCompose { etid ->
-            entityTypes.getAsync(etid!!)
-        }.thenApplyAsync { entityType ->
-            entityType.properties.mapTo(mutableSetOf()) { ptid ->
-                AccessCheck(AclKey(entitySetId, ptid), EnumSet.of(Permission.READ))
-            }
-        }
-
-        val transporterColumnsCompletion = entityTypeId.thenCompose { etid ->
+        ).thenCompose { etid ->
             requireNotNull(etid) {
                 "Entity set $entitySetId has no entity type"
             }
             transporterState.submitToKey(etid, GetPropertyTypesFromTransporterColumnSetEntryProcessor())
-        }.thenCompose { transporterPtIds ->
+        }
+
+        val ptIdsToFqns = ptIds.thenCompose { transporterPtIds ->
             propertyTypes.submitToKeys(transporterPtIds, GetFqnFromPropertyTypeEntryProcessor())
+        }.thenApply { ptIdToFqn ->
+            ptIdToFqn.mapTo( mutableSetOf() ) { PropertyTypeIdFqn(it.key, it.value) }
         }
 
-        val userToPermissionsCompletion = accessCheckCompletion.thenApply { accessChecks ->
-            userToPrincipals.mapValues { (_, principals) ->
-                authorizationManager.accessChecksForPrincipals(accessChecks, principals).filter {
-                    it.permissions[Permission.READ]!!
-                }.map {
-                    it.aclKey[1]
-                }.toList()
+        val tableCols = ptIds.thenApply { transporterPtIds ->
+            transporterPtIds.associate {
+                AclKey(entitySetId, it) to TableColumn(organizationId, entitySetId, it)
             }
         }
 
-        userToPermissionsCompletion.thenCombine(transporterColumnsCompletion) { userToPtCols, transporterColumns ->
-            val userToEntitySetColumnNames = userToPtCols.mapValues { (_, columns) ->
-                columns.map {
-                    transporterColumns.get(it).toString()
-                }
+        val acls = permissions.entrySet(Predicates.and(
+                Predicates.equal<AceKey, AceValue>(PermissionMapstore.ROOT_OBJECT_INDEX, entitySetId),
+                Predicates.equal<AceKey, AceValue>(PermissionMapstore.SECURABLE_OBJECT_TYPE_INDEX, SecurableObjectType.PropertyTypeInEntitySet))
+        ).map { (key, valu) ->
+            Acl( key.aclKey, listOf( Ace(key.principal, valu as Set<Permission>, Optional.of(valu.expirationDate))) )
+        }
+
+        ptIdsToFqns.thenCombine( tableCols ) { asPtFqns, colsById ->
+            try {
+                entitySets.lock(entitySetId, 10, TimeUnit.SECONDS)
+                val es = entitySets.getValue(entitySetId)
+                transporterService.assembleEntitySet(
+                        organizationId,
+                        es,
+                        asPtFqns,
+                        acls,
+                        colsById
+                )
+                es.flags.add(EntitySetFlag.TRANSPORTED)
```

Comment:
I thought we had like an AddFlagsToEntitySetEntryProcessor kind of thing

Reply:
We do, but we're locking the entityset here and have already read it out so I thiiink doing a set here is the way to do it. It was unclear to me whether AddFlagsToEntitySetEP works here actually. Happy to swap for that though cause it would save serialization cost on the way to PG

Reply:
updated to EP